src/VisualStudio/Core/Def/Implementation/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -478,16 +583,16 @@ private bool IsLive(Project project, Compilation compilation, DiagnosticData dia
                     return false;
                 }
 
-                if (SupportedLiveDiagnosticId(project, compilation, diagnosticData.Id))
+                if (await IsSupportedLiveDiagnosticIdAsync(project, diagnosticData.Id).ConfigureAwait(false))
                 {
                     return true;
                 }
```

Comment:
i basically skimmed up to here. this is a really big and complex state-machine/data-flow system.  it would be really helpful to really explain what's going on through all of this to get a sense for what the state of hte system is and why the work we're doing is appropriate.

Reply:
I have added a top-level doc comment to `ExternalErrorDiagnosticUpdateSource` type declaration, hope that is useful.